torchmultimodal/models/video_vqvae.py

```diff
@@ -14,47 +14,60 @@
 from torchmultimodal.utils.assertion import assert_equal_lengths
 from torchmultimodal.utils.common import to_tuple_tuple
 
-DEFAULT_ENCODER_IN_CHANNEL_DIMS = (3, 240, 240, 240, 240, 240)
-DEFAULT_DECODER_OUT_CHANNEL_DIMS = (240, 240, 240, 240, 3)
-
 
 def video_vqvae(
-    encoder_in_channel_dims: Tuple[int, ...] = DEFAULT_ENCODER_IN_CHANNEL_DIMS,
-    encoder_kernel_sizes: int = 4,
-    encoder_strides: int = 2,
+    in_channel_dim: int = 3,
+    encoder_hidden_dim: int = 240,
+    encoder_kernel_size: int = 3,
+    encoder_stride: int = 2,
+    encoder_n_layers: int = 1,
     n_res_layers: int = 4,
     attn_hidden_dim: int = 240,
-    num_embeddings: int = 2048,
+    num_embeddings: int = 1024,
     embedding_dim: int = 256,
-    decoder_out_channel_dims: Tuple[int, ...] = DEFAULT_DECODER_OUT_CHANNEL_DIMS,
-    decoder_kernel_sizes: int = 4,
-    decoder_strides: int = 2,
+    decoder_hidden_dim: int = 240,
+    decoder_kernel_size: int = 4,
+    decoder_stride: int = 2,
+    decoder_n_layers: int = 1,
 ) -> VQVAE:
-    """Construct Video VQVAE with default parameters used in VideoGPT (Yan et al. 2022). Code ref:
+    """Generic Video VQVAE builder using default parameters from VideoGPT (Yan et al. 2022).
+    Uses hyperparameters from Appendix A Table 8 for BAIR/RoboNet/ViZDoom. Code ref:
     https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
```

Comment:
I think the only delta between mugen and the original videoGPT is the kernel size. Unless I'm missing something (if videogpt overrides their default parameters somewhere).

Reply:
Ignore this for now. i wasn't looking at Appendix earlier.